### ISSUE TITLE
Bug? mintty + vim w/ `mouse=a`

### ISSUE THREAD

**Issue by keighrim** [mintty/mintty#782]
I'm using terminal vim with mintty as the terminal emulator (msys is my main bash for Windows). When I move the window focus by a mouse click to mintty running vim (from any other app), in additional to the focus moving to the mintty window, vim scrolls three lines up (I think it's doing literally "3&lt;c-y&gt;"), instead of moving cursor to where I clicked in the buffer. However, when I'm running vim _inside tmux_, only the focus moves and nothing else happen. 
I went through my vimrc and found that `mouse=X` (a, n, v any option) is causing this. 
Is this some kind of bug in mintty? 

(The same happens when I'm ssh-ing to OSX or linux machines using mintty + openssh)

**Comment by mintty** [comment]
Hold on; it looks as if a mouse release event is sent initially when focussing the window. This might be confusing vim, further analysis needed.

**Comment by mintty** [comment]
Mintty 2.8.2 introduced a new feature to prevent accidental text selection on mouse-click window activation (#717) by suppressing the first mouse event after focussing a window. However, it fails to suppress the mouse release event too. While this is subject to improvement, vim should not interpret a singular mouse release event as a scroll-down action.

**Comment by mintty** [comment]
Released 2.9.1.

**Comment by ahsanu123** [comment]
hey, iam newbie about nvim and mintty.
it this issus fix? how i setting mintty/neovim so i can enable mouse functionality??

i try to enable mouse function on neovim, but not working. please help.
thanks.

**Comment by mintty** [comment]
The issue above was fixed with 2.9.1.
I am not a vim user myself but I once set up a `.vimrc` file which looks like this:
```
set mouse=a
if has("mouse_sgr")
    set ttymouse=sgr
else
    set ttymouse=xterm2
end
